[bug] jinja2 minimum version required it not correct

### Environment Details (include every applicable attribute)
  * Operating System+version: Ubuntu 18
  * Compiler+version: gcc-7
  * Conan version: 1.29.1
  * Python version: 3.7

### Steps to reproduce (Include if Applicable)
While upgrading conan to a recent version, we get error due to an out of date jinja2 version on the system.
Reproducible by forcing jinja2 version < 2.9

`     [exec]   File "/usr/local/lib/python3.7/dist-packages/conans/client/cache/cache.py", line 7, in <module>
     [exec]     from jinja2 import Environment, select_autoescape, FileSystemLoader, ChoiceLoader
     [exec] ImportError: cannot import name 'select_autoescape' from 'jinja2' (/usr/lib/python3/dist-packages/jinja2/__init__.py)
     [exec]     from jinja2 import Environment, select_autoescape, FileSystemLoader, ChoiceLoader
     [exec] ImportError: cannot import name 'select_autoescape' from 'jinja2' (/usr/lib/python3/dist-packages/jinja2/__init__.py)
     [exec] -- Configuring incomplete, errors occurred!
     [exec] See also "/tmp/src/k0/tools/generated/CMakeFiles/CMakeOutput.log".
     [exec] CMake Error at generated/conan.cmake:328 (message):
     [exec]   Conan install failed='1'`

select_autoescape appears to have been added in jinja2 2.9. Conan's requirements.txt only specifies 2.3 as a minimum. 

https://github.com/conan-io/conan/blob/47928cbcd3db1ff05798c5f36fc900412f9fa7c9/conans/requirements.txt#L15

--- a/conans/requirements.txt
+++ b/conans/requirements.txt
@@ -12,7 +12,7 @@ future>=0.16.0, <0.19.0
 pygments>=2.0, <3.0
 deprecation>=2.0, <2.1
 tqdm>=4.28.1, <5
-Jinja2>=2.3, <3
+Jinja2>=2.9, <3
 python-dateutil>=2.7.0, <3
 idna==2.6 ; sys_platform == "darwin" # Solving conflict, somehow is installing 2.7 when requests require 2.6
 cryptography>=1.3.4, <2.4.0 ; sys_platform == "darwin"

